### DISCUSSION TITLE
Multiple instead of double exposure

### DISCUSSION THREAD

**Discussion by Hazoooka** [sn4k3/UVtools#778] [Ideas]
There is a very handy tool called "double exposure" that can do some really good stuff.
However, lately I have found it necessary to add at least one more exposure run.
That would make triple exposure.
But I bet it's also possible to bump at the situation when a quadruple exposure would be necessary.

Running "double exposure" twice does the job of course, but it would be more straight forward to be able to choose the number of additional exposures in the menu.

Regards,
Witold.

**Comment by sn4k3** [comment]
What are the use cases for triple or more exposures?
Please elaborate the scenario so I can investigate it.

**Comment by Hazoooka** [comment]
I use double exposure for differentiating the antialliasing, but also for the elephant foot.

First pass is for blurred objects, smooth surfaces.
Second pass is for the details.
There is another pass needed when those two coexist on the flat platform, large enough to get warped, or on the first layers where I also need to fight the elephant foot.
So the extra pass is for long time exposure for the core of the platform to make it strong in advance.

Those platforms in my case are pretty thin, 0.2mm or 0.3mm but it makes 10 or 30 extra layers depending on the layer height.
Details with smooth surfaces are merged after both files are processed initially, all the necessary parameters tweaked on both CTBs the same way. Only then importing is successful, all the shapes fit together.

Double exposure is sometimes applied several times on an object.
Using the function carefully one can succeed.
Adding the extra double exposure after the first one makes the whole process more difficult as every next exposure needs to be added manually remembering which layers to treat extra.

It’s all doable but time consuming and easy to make mistakes.
Having a choice for how many different exposures we can apply will help organising all this better.

**Comment by Hazoooka** [comment]
To complement the description I wish to add that projects may require several different exposures for different types of elements on the object: thinner, thicker, hanging firmly attached... etc.
So flexible number of exposure passes is a good idea.

**Reply by Hazoooka** [reply]
I have been further thinking agout this and it seems like actually 4 exposure passes should be enough for most applications.

**Reply by sn4k3** [reply]
I will look a way to make it possible to customize the passes.

**Comment by Hazoooka** [comment]
Thank you Tiago.
This should do the job :)

**Comment by Hazoooka** [comment]
Hi Tiago.


I have finally come to test the new phased exposure option, took me some time I agree.
I can't seem to asign different slices to different exposure runs.
This was actually the goal of this new feature, to be able to prolong exposure on separate parts of the model in separate sets of slices.

Would it be possible to make it work?

Kind regards,
Witold.

**Comment by Hazoooka** [comment]
Alright, I remember now, time took it's toll apparently.

However, the question stands.
If I want to use 3 passes with different parts of the model I slice 3 separate CTBs.
Now, to sign those 3 CTBs to 3 passes I need to expand each CTB by 3 with the right slices erased, and then import them accordingly in place into the main stack, starting from the proper layer each time.
It's a tedious process with a lot of room for error.

Would it be possible to simplify it by importing separate CTB to the separate pass?
That would be really simple and time saving.

Kind regards,
Witold.

**Comment by Hazoooka** [comment]
By importing different CTBs I mean to import a set of slices.
For those not accustomed with importing the slices, it's brilliant, we do not have to make actual PNG slices, we can directly get them from the CTB.

Additionally, it would be good to have a layer range for each pass selected in the pass menu, not for the whole file.
Each pass menu can also have the slice set import selection.
Everything can be saved in the project file, with a possibility to easily change each pass if necessary.
The final compilation on save or export will do the math.

Kind regards,
Witold.
